compiler-core/src/error.rs

```diff
@@ -149,7 +151,11 @@ pub enum Error {
     Gzip(String),
 
     #[error("shell program `{program}` not found")]
-    ShellProgramNotFound { program: String },
+    ShellProgramNotFound {
+        program: String,
+        os: OS,
+        distro: Distro,
```

Comment:
`Distro` is only valid if `OS` is `Linux`, so it should be a value held within the `Linux` variant of `OS`.

For example, this is currently representable but invalid

```
os: Windows,
distro: Debian
```

Reply:
That makes sense, I moved Distro inside Linux as `Linux(Distro)` and get_os() will contain info about both `OS` and `Distro`(under linux)